### ISSUE TITLE
Cannot access user-defined derived arrays

### ISSUE THREAD

**Issue by ibackus** [pynbody/pynbody#250]
I'm trying for the first time to define a derived array, but it doesn't work the way I expect it to according to [this tutorial](http://pynbody.github.io/pynbody/tutorials/data_access.html#create-arrays).  

Here's how I define the array:
```python
@pynbody.derived_array
def exists(sim):
    
    return (sim['N'] > 0)
```

Then I try to access the array in a snapshot:
```python
f['exists']
Traceback (most recent call last):

  File "<ipython-input-487-9a9d1d6ec300>", line 1, in <module>
    f['exists']

  File "/usr/local/lib/python2.7/dist-packages/pynbody/snapshot/__init__.py", line 289, in __getitem__
    return self._get_array(i)

  File "/usr/local/lib/python2.7/dist-packages/pynbody/snapshot/__init__.py", line 1193, in _get_array
    x = self._arrays[name]

KeyError: 'exists'
```

Which is surprising, because 'exists' shows up in the derivable keys:
```python
'exists' in f.derivable_keys()
Out[488]: True
```

Any ideas?

**Comment by apontzen** [comment]
Did you define the array before you loaded the simulation? I think (from memory) that order might matter, but then of course it shouldn't show up in derivable_keys if so... Sounds like a bug one way or another.

**Comment by apontzen** [comment]
Another possibility is that your exists function is throwing an exception which pynbody then interprets as being 'underivable' - can you check by running the function directly on the snapshot object?

**Comment by ibackus** [comment]
So I did a bit of poking around and here's what I think was happening.  I was creating a new SimSnap on my own (with using pynbody.new or pynbody.load) because I was doing some kinda non-standard stuff and I didn't want to use the families.  So my SimSnaps had no families defined, which appears to have lead to an error when trying to access arrays.  When I created a SimSnap with star particles instead it worked.
